src/estimagic/optimization/tranquilo/fit_models.py

```diff
@@ -233,16 +233,114 @@ def _fit_ridge(x, y, penalty):
 
     Returns:
         np.ndarray: Array of shape (p, k) of coefficients.
-
     """
     a = x.T @ x
     b = x.T @ y
 
     coef, *_ = np.linalg.lstsq(a + np.diag(penalty), b, rcond=None)
     coef = coef.T
+
     return coef
 
 
+def fit_pounders(x, y, model_info):
+    """Fit a linear model using the pounders fitting method.
+
+    The solution represents the quadratic whose Hessian matrix is of
+    minimum Frobenius norm.
+
+    For a mathematical exposition, see :cite:`Wild2008`, p. 3-5.
+
+    Args:
+        x (np.ndarray): Array of shape (n_samples, n_params) of x-values,
+            rescaled such that the trust region becomes a hypercube from -1 to 1.
+        y (np.ndarray): Array of shape (n_samples, n_residuals) with function
+            evaluations that have been centered around the function value at the
+            trust region center.
+        model_info (ModelInfo): Information that describes the functional form of the
+            model.
+
+    Returns:
+        np.ndarray: The model coefficients.
+    """
+    n_samples, n_params = x.shape
+    has_intercepts = model_info.has_intercepts
+    has_squares = model_info.has_squares
+
+    features = _polynomial_features(x, has_intercepts, has_squares)
+    m_mat, m_mat_pad, n_mat = _build_feature_matrices_pounders(
+        features, n_params, n_samples, has_intercepts
+    )
+    z_mat = _calculate_basis_null_space(m_mat_pad, n_samples, n_params)
+    n_z_mat = _multiply_feature_matrix_with_basis_null_space(
+        n_mat, z_mat, n_samples, n_params
+    )
+
+    coef = _get_current_fit_pounders(
+        y, m_mat, n_mat, z_mat, n_z_mat, n_samples, n_params, has_intercepts
+    )
+
+    return coef
+
+
+def _get_current_fit_pounders(
+    y, m_mat, n_mat, z_mat, n_z_mat, n_samples, n_params, has_intercepts
+):
+    n_residuals = y.shape[1]
+
+    n_poly_features = n_params * (n_params + 1) // 2
+    offset = 0 if has_intercepts else 1
+
+    coef = np.empty((n_residuals, has_intercepts + n_params + n_poly_features))
+
+    # just-identified case
+    if n_samples == (n_params + 1):
+        omega = np.zeros(n_params)
```

Comment:
I think we need better names for omega, beta and alpha. Rest can stay close to the paper, but these three seem to be fundamental to understand what's going on.